libsolidity/analysis/TypeChecker.cpp

```diff
@@ -587,11 +587,14 @@ bool TypeChecker::visit(VariableDeclaration const& _variable)
 	if (isStructMemberDeclaration)
 		return false;
 
-	if (auto referenceType = dynamic_cast<ReferenceType const*>(varType))
+	auto referenceType = dynamic_cast<ReferenceType const*>(varType);
+	auto mappingType = dynamic_cast<MappingType const*>(varType);
+	if (referenceType || mappingType)
 	{
-		auto result = referenceType->validForLocation(referenceType->location());
+		DataLocation loc = referenceType? referenceType->location(): DataLocation::Storage;
+		auto result = varType->validForLocation(loc);
```

Comment:
```suggestion
		BoolResult result = varType->validForLocation(referenceType? referenceType->location() : DataLocation::Storage);
```
A few different things here:
1) `loc` is used only once and does not have a good name so maybe just inline it.
2) Return type is not obvious (I'd actually say `bool` if I had to guess) so I'd not use `auto`.
3) Space before `:` in ternary operator.

Reply:
OK